kubectl-cloudflow/verify/blueprint_test.go

```diff
@@ -0,0 +1,259 @@
+			cdr-aggregator.out = [console-egress.in]
+		}
+	}`
+}
+
+func createBlueprintSampleWithImageInStreamletNotPresentInImages() string {
+	return `blueprint {
+		name = call-record-aggregator
+		images {
+			aggr = "lightbend/spark-aggregation:134-d0ec286-dirty"
+			ings = "lightbend/akka-cdr-ingestor:134-d0ec286-dirty"
+			outp = "lightbend/akka-java-aggregation-output:134-d0ec286-dirty"
+		}
+		streamlets {
+			cdr-generator1 = notInImage/carly.aggregator.CallRecordGeneratorIngress
+			cdr-generator2 = notInImage/carly.aggregator.CallRecordGeneratorIngress
+			merge = ings/carly.ingestor.CallRecordMerge
+			cdr-ingress = ings/carly.ingestor.CallRecordIngress
+			cdr-aggregator = aggr/carly.aggregator.CallStatsAggregator
+			console-egress = outp/carly.output.AggregateRecordEgress
+			error-egress = outp/carly.output.InvalidRecordEgress
+		}
+		connections {
+			cdr-generator1.out = [merge.in-0]
+			cdr-generator2.out = [merge.in-1]
+			cdr-ingress.out = [merge.in-2]
+			merge.valid = [cdr-aggregator.in]
+			merge.invalid = [error-egress.in]
+			cdr-aggregator.out = [console-egress.in]
+		}
+	}`
+}
+
+func createBlueprintSampleWithConnectionsHavingInvalidStreamlets() string {
+	return `blueprint {
+		name = call-record-aggregator
+		images {
+			aggr = "lightbend/spark-aggregation:134-d0ec286-dirty"
+			ings = "lightbend/akka-cdr-ingestor:134-d0ec286-dirty"
+			outp = "lightbend/akka-java-aggregation-output:134-d0ec286-dirty"
+		}
+		streamlets {
+			cdr-generator1 = aggr/carly.aggregator.CallRecordGeneratorIngress
+			cdr-generator2 = aggr/carly.aggregator.CallRecordGeneratorIngress
+			merge = ings/carly.ingestor.CallRecordMerge
+			cdr-ingress = ings/carly.ingestor.CallRecordIngress
+			cdr-aggregator = aggr/carly.aggregator.CallStatsAggregator
+			console-egress = outp/carly.output.AggregateRecordEgress
+			error-egress = outp/carly.output.InvalidRecordEgress
+		}
+		connections {
+			generator1.out = [merge.in-0]
+			generator2.out = [merge.in-1]
+			cdr-ingress.out = [merge.in-2]
+			merge.valid = [cdr-aggregator.in]
+			merge.invalid = [error-egress.in]
+			cdr-aggregator.out = [console-egress.in]
+		}
+	}`
+}
+
+func Test_VerifyConnectionHash(t *testing.T) {
+	vInlet := VerifiedInlet{
+		VerifiedPort{
+			"blabla",
+			domain.InOutletSchema{
+				"adadasdasda",
+				"{\"type\":\"record\",\"name\":\"Data\",\"namespace\":\"sensors\",\"fields\":[{\"name\":\"src\",\"type\":\"string\"},{\"name\":\"timestamp\",\"type\":\"long\"},{\"name\":\"gauge\",\"type\":\"string\"},{\"name\":\"value\",\"type\":\"double\"}]}",
+				"",
+				"avro",
+			},
+		},
+		VerifiedStreamlet{},
+	}
+
+	hash1 := GetSHA256Hash(vInlet)
+
+	vEmptyInlet := VerifiedInlet{}
+	hash2 := GetSHA256Hash(vEmptyInlet)
+	assert.NotEqual(t, hash1, hash2)
+}
+
+func Test_VerifyValidBlueprint(t *testing.T) {
+	errors := VerifyBlueprint(createValidBlueprintSample())
+	assert.Empty(t, errors)
+}
+
+func Test_VerifyBlueprintWithoutImages(t *testing.T) {
```

Comment:
@RayRoestenburg this is the test for having no images in the blueprint. @debasishg already has one. 
Btw the expression `images := config.GetNode("blueprint.images").GetObject().Items()` will not fail because in that framework they alwas check for nil. Pls check for example https://github.com/go-akka/configuration/blob/master/hocon/value.go#L201